ocaml/xapi/xapi_http.ml

```diff
@@ -155,7 +161,15 @@ let assert_credentials_ok realm ?(http_action = realm) ?(fn = Rbac.nofn)
         raise (Failure (Printf.sprintf "Unknown authorization header: %s" x))
     | None, None, None ->
         debug "No header credentials during http connection to %s" realm ;
-        raise (Http.Unauthorised realm)
+        let session_id =
+          try create_session_for_client_cert req ic
```

Comment:
Let's add a comment and some debug logging here to avoid confusion.
```suggestion
          (* The connection may have been authenticated using a client cert.
               If so, then following call confirms this. *)
          try
            create_session_for_client_cert req ic;
            debug "This HTTP connection was pre-authenticated using a client cert"
```